controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/filter/SignatureFilterTest.java

```diff
@@ -1,55 +1,100 @@
+                                                 "-----END PUBLIC KEY-----\n";
+
     private static final String privateKey = "-----BEGIN EC PRIVATE KEY-----\n" +
                                                   "MHcCAQEEIJUmbIX8YFLHtpRgkwqDDE3igU9RG6JD9cYHWAZii9j7oAoGCCqGSM49\n" +
                                                   "AwEHoUQDQgAEuKVFA8dXk43kVfYKzkUqhEY2rDT9z/4jKSTHwbYR8wdsOSrJGVEU\n" +
                                                   "PbS2nguIJ64OJH7gFnxM6sxUVj+Nm2HlXw==\n" +
                                                   "-----END EC PRIVATE KEY-----\n";
 
+    private static final ApplicationId id = ApplicationId.from("my-tenant", "my-app", "default");
+
+    private ControllerTester tester;
+    private ApplicationController applications;
+    private SignatureFilter filter;
+    private RequestSigner signer;
+
+    @Before
+    public void setup() {
+        tester = new ControllerTester();
+        applications = tester.controller().applications();
+        filter = new SignatureFilter(tester.controller());
+        signer = new RequestSigner(privateKey, id.serializedForm(), tester.clock());
+
+        tester.createApplication(tester.createTenant(id.tenant().value(), "unused", 496L),
+                                 id.application().value(),
+                                 id.instance().value(),
+                                 28L);
+    }
+
     @Test
-    public void testResponses() {
-        ContainerTester tester = new ContainerTester(container, responseFiles);
-        ApplicationId id = ApplicationId.from("my-tenant", "my-app", "default");
-        Optional<Credentials> credentials = Optional.of(new Credentials(() -> "user"));
-
-        // Create an application.
-        tester.controller().tenants().create(new CloudTenantSpec(TenantName.from("my-tenant"), "token"), credentials.get());
-        tester.controller().applications().createApplication(id, credentials);
-
-        // PATCH in a pem deploy key.
-        tester.assertResponse(request("/application/v4/tenant/my-tenant/application/my-app", PATCH)
-                                      .roles(Set.of(Role.hostedOperator()))
-                                      .data("{\"pemDeployKey\":\"" + publicKey + "\"}"),
-                              "{\"message\":\"Set pem deploy key to " +
-                              publicKey.replaceAll("\\n", "\\\\n") + "\"}");
-
-        tester.assertResponse(request("/application/v4/tenant/my-tenant/application/my-application/instance/default/submit", POST)
-                             .roles());
+    public void testFilter() {
+        // Unsigned request is rejected.
+        HttpRequest.Builder request = HttpRequest.newBuilder(URI.create("https://host:123/path/./..//..%2F?query=empty&%3F=%26"));
+        byte[] emptyBody = new byte[0];
+        DiscFilterRequest unsigned = requestOf(request.method("GET", HttpRequest.BodyPublishers.ofByteArray(emptyBody)).build(), emptyBody);
```

Comment:
I have better experience of just mocking `DiscFilterRequest` directly with Mockito. Wrapping multiple request types is brittle, and `ApplicationRequestToDiscFilterRequestWrapper` is more of an ad-hoc hack.

Reply:
Will consider that in the future.